// angular-animations/esm5/attention-seekers/heart-beat.animation.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { animate, animateChild, animation, AUTO_STYLE, group, keyframes, query, style, transition, trigger, useAnimation } from '@angular/animations';
/**
 * @record
 */
export function IHeartBeatAnimationOptions() { }
if (false) {
    /**
     * Scale factor
     *
     * Default: 1.3
     * @type {?|undefined}
     */
    IHeartBeatAnimationOptions.prototype.scale;
}
/** @type {?} */
var heartBeat = animation([
    animate('{{duration}}ms {{delay}}ms', keyframes([
        style({ visibility: AUTO_STYLE, transform: 'scale(1)', easing: 'ease-in-out', offset: 0 }),
        style({ transform: 'scale({{scale}})', easing: 'ease-in-out', offset: 0.14 }),
        style({ transform: 'scale(1)', easing: 'ease-in-out', offset: 0.28 }),
        style({ transform: 'scale({{scale}})', easing: 'ease-in-out', offset: 0.42 }),
        style({ transform: 'scale(1)', easing: 'ease-in-out', offset: 0.7 })
    ]))
]);
/** @type {?} */
var DEFAULT_DURATION = 1300;
/** @type {?} */
var DEFAULT_SCALE = 1.3;
/**
 * @param {?=} options
 * @return {?}
 */
export function heartBeatAnimation(options) {
    return trigger((options && options.anchor) || 'heartBeat', [
        transition("0 " + ((options && options.direction) || '<=>') + " 1", tslib_1.__spread((options && options.animateChildren === 'before' ? [query('@*', animateChild(), { optional: true })] : []), [
            group(tslib_1.__spread([
                useAnimation(heartBeat)
            ], (!options || !options.animateChildren || options.animateChildren === 'together'
                ? [query('@*', animateChild(), { optional: true })]
                : [])))
        ], (options && options.animateChildren === 'after' ? [query('@*', animateChild(), { optional: true })] : [])), {
            params: {
                delay: (options && options.delay) || 0,
                duration: (options && options.duration) || DEFAULT_DURATION,
                scale: (options && options.scale) || DEFAULT_SCALE
            }
        })
    ]);
}
/**
 * @param {?=} options
 * @return {?}
 */
export function heartBeatOnEnterAnimation(options) {
    return trigger((options && options.anchor) || 'heartBeatOnEnter', [
        transition(':enter', tslib_1.__spread((options && options.animateChildren === 'before' ? [query('@*', animateChild(), { optional: true })] : []), [
            style({ visibility: 'hidden' }),
            group(tslib_1.__spread([
                useAnimation(heartBeat)
            ], (!options || !options.animateChildren || options.animateChildren === 'together'
                ? [query('@*', animateChild(), { optional: true })]
                : [])))
        ], (options && options.animateChildren === 'after' ? [query('@*', animateChild(), { optional: true })] : [])), {
            params: {
                delay: (options && options.delay) || 0,
                duration: (options && options.duration) || DEFAULT_DURATION,
                scale: (options && options.scale) || DEFAULT_SCALE
            }
        })
    ]);
}
